libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,184 @@
+        self.csm_conf = configmanager.get_config_wrapper(
+            fpath="config/csm/test_rest_query_deployment.yaml")
+        self.local_sol_path = LOCAL_SOLUTION_PATH
+        self.resp_dix = None
+        self.resp_sns = None
+        self.prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        self.csm_obj = csm_api_factory("rest")
+
+    def get_solution_yaml(self):
+        """
+        Read solution yaml into a dictionary
+        """
+        remote_sol_path = self.prov_deploy_cfg["git_remote_path"] + "solution.example.yaml"
+        solution_path = self.csm_obj.master.copy_file_to_local(remote_path=remote_sol_path,
+                                                               local_path=self.local_sol_path)
+        data = yaml.safe_load(solution_path)
+        return data
+
+    @RestTestLib.authenticate_and_login
+    def get_system_topology(self, auth_header=None, query_param=None):
+        """
+        Get system topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get system topology request....")
+        if auth_header is None:
+            headers = self.headers
+        else:
+            headers = auth_header
+        endpoint = self.config["get_system_topology"]
+        if query_param is not None:
+            endpoint = endpoint + '/' + query_param
+        response = self.restapi.rest_call("get", endpoint=endpoint,
+                                          headers=headers)
+        self.log.info("Get system topology request successfully sent...")
+        return response
+
+    # Function not ready
+    def verify_system_topology(self, expected_response=HTTPStatus.OK):
+        """
+        Verify Get system topology
+        """
+        resp = self.get_system_topology()
+        result = True
+        if resp.status_code == expected_response:
+            result = True
+            get_response = resp.json()
+            self.log.info("Verify id fields have unique values")
+            self.log.info("Verfying cvg and devices in resp and solution yaml")
+            # Verify deployment version and deployment time
+        else:
+            self.log.error("Status code check failed.")
+            result = False
+        return result, get_response
+
+    @RestTestLib.authenticate_and_login
+    def get_cluster_topology(self, auth_header=None, cluster_id=None):
+        """
+        Get cluster topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get cluster topology request....")
+        query_param = 'cluster'
+        if cluster_id is not None:
+            query_param = query_param + '/' + cluster_id
+        response = self.get_system_topology(auth_header, query_param)
+        return response
+
+    @RestTestLib.authenticate_and_login
+    def get_storage_topology(self, auth_header=None, storage_set_id=None):
```

Comment:
Cluster id should be required parameter here

Reply:
done